### ISSUE TITLE
SSL CDN // Do not redirect to unencrypted connection 

### ISSUE THREAD

**Issue by rbu** [voc/media.ccc.de#25]
I was surprised to find out that I can browse the media.ccc.de site with HTTPS, and I can browse the cdn.media.ccc.de directory listings with SSL just fine. It even contains HTTP headers to upgrade to SSL when using an unencrypted connection.
However, when downloading a file, I am automatically redirected to HTTP:
```
HTTP/1.1 302 Found
Connection: keep-alive
Content-Length: 307
Content-Type: text/html; charset=iso-8859-1
Date: Fri, 01 Jan 2016 21:25:04 GMT
Digest: MD5=HCGw5mREf+goke3ouID1iQ==
Digest: SHA=H1jYHHF8Vj0DJybeetHcnZWm9S4=
Digest: SHA-256=DHE74iN9zjeZrxpUuDEgxkWyPUU0lGrXsr/TESFfYdk=
Link: <http://cdn.media.ccc.de/congress/2015/h264-hd/32c3-7331-en-de-The_exhaust_emissions_scandal_Dieselgate_hd.mp4.meta4>; rel=describedby; type="application/metalink4+xml"
Link: <http://cdn.media.ccc.de/congress/2015/h264-hd/32c3-7331-en-de-The_exhaust_emissions_scandal_Dieselgate_hd.mp4.torrent>; rel=describedby; type="application/x-bittorrent"
Link: <http://ftp.halifax.rwth-aachen.de/ccc/congress/2015/h264-hd/32c3-7331-en-de-The_exhaust_emissions_scandal_Dieselgate_hd.mp4>; rel=duplicate; pri=1; geo=de
Link: <http://mirror.netcologne.de/CCC/congress/2015/h264-hd/32c3-7331-en-de-The_exhaust_emissions_scandal_Dieselgate_hd.mp4>; rel=duplicate; pri=2; geo=de
Link: <http://mirror.eu.oneandone.net/projects/media.ccc.de/congress/2015/h264-hd/32c3-7331-en-de-The_exhaust_emissions_scandal_Dieselgate_hd.mp4>; rel=duplicate; pri=3; geo=de
Link: <http://mirror-1.server.selfnet.de/CCC/congress/2015/h264-hd/32c3-7331-en-de-The_exhaust_emissions_scandal_Dieselgate_hd.mp4>; rel=duplicate; pri=4; geo=de
Link: <http://ftp.fau.de/cdn.media.ccc.de/congress/2015/h264-hd/32c3-7331-en-de-The_exhaust_emissions_scandal_Dieselgate_hd.mp4>; rel=duplicate; pri=5; geo=de
Location: http://ftp.halifax.rwth-aachen.de/ccc/congress/2015/h264-hd/32c3-7331-en-de-The_exhaust_emissions_scandal_Dieselgate_hd.mp4
Server: nginx/1.4.6 (Ubuntu)
Strict-Transport-Security: max-age=15768000
X-AS: 131279
X-MirrorBrain-Mirror: halifax
X-MirrorBrain-Realm: region
X-Prefix: 175.45.176.0/24
```

Currently, 2 out of your 12 mirrors serve files via HTTPS (fau.de and nusedu). Hopefully this ratio is going to increase over the next time. I can use the encrypted mirrors manually for now, but it would be nice to stay on HTTPS with the web player as well.

So in this ticket I propose to redirect HTTPS CDN requests to HTTPS mirros.

Edit: I realize the fact that you send HTTPS upgrade headers would render all mirrors with HTTP-only unused.

**Comment by manno** [comment]
This is a known issue, but we should document it on the [about page](https://media.ccc.de/about.html). Right below the privacy policy:  "Video and audio files are often provided by mirrors and are not covered by this policy."

The CDN logic is not implemented by this project, but by http://mirrorbrain.org/
As far as I know redirecting to HTTPS is currently not possible. 

Since video files are only [passive content](https://developer.mozilla.org/en/docs/Security/MixedContent) this is not a priority for me.

**Comment by rbu** [comment]
For me personally, using HTTPS for playback increases confidentiality with a network-level attacker. I'd feel strange watching one of the political talks when in an authoritarian regime's network. What's the point of serving the rest of the site via HTTPS at all, as you don't provide credentials?

For what it's worth, Mirrorbrain implemented this feature in the latest release, 2.19.0 according to http://mirrorbrain.org/issues/issue131. Ref http://mirrorbrain.org/files/releases/

**Comment by manno** [comment]
I understand your privacy concerns, but I don't think issue131 implements the necessary functionality.
First mirrorbrain would need to know if HTTPS is possible for every configured mirror.  Then it would need to detect if HTTPS was actually requested, while running behind a proxy.

We're currently evaluating https://github.com/etix/mirrorbits and might switch in 2016. But currently other issues are more important to me.
